staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -839,6 +839,7 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 
 			MaxRequestBodyBytes: r.maxRequestBodyBytes,
 		}
+		isSubresource := subresources != nil
```

Comment:
I don't know this particular code-- does the main resource go through this code path? I would expect the main resource and subresources to come up with different values for this.

Reply:
>  does the main resource go through this code path?
Yes, from what I understand this is the case.

Reply:
Interesting, I need to investigate this path a bit more. Thanks for raising this.

Reply:
Basically CRDs have the main resource which uses a `FieldManager` [1], the status sub-resource that uses the same FieldManager [3], and the scale subresource that uses no manager [4]. We clearly can't use the same FieldManager for main resource and status sub-resource so we'll have to create a new one that has the "ignore" flag explicitly set to true.

[1]: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go#L844-L852
[2]: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go#L878
[3]: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go#L873

Reply:
I have fixed the problem by adding a separate `FieldManager` for the status subresource (https://github.com/kubernetes/kubernetes/pull/91993/commits/02c1948f71de628628034caf8c6d98bd1816ec5c). Given `scale` doesn't have a `FieldManager` yet I have left it as it is.